internal/grpc_service_config/config.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package grpc_service_config
+
+import (
+	"log"
+
+	"github.com/golang/protobuf/ptypes"
+	duration "github.com/golang/protobuf/ptypes/duration"
+)
+
+// Config represents parsed mapping of the gRPC ServiceConfig contents
+// with methods for increased accessibility.
+type Config struct {
+	policies  map[string]*MethodConfig_RetryPolicy
+	timeouts  map[string]*duration.Duration
+	reqLimits map[string]int
+	resLimits map[string]int
+}
+
+// New traverses the given gRPC ServiceConfig into more accessible constructs
+// mapped by the names to the specific config values. Use the accessors on the
+// resulting Config to retrieve values for a Service or Method.
+func New(c ServiceConfig) Config {
+	policies := map[string]*MethodConfig_RetryPolicy{}
+	timeouts := map[string]*duration.Duration{}
+	reqLimits := map[string]int{}
+	resLimits := map[string]int{}
+
+	// gather retry policies from MethodConfigs
+	for _, mc := range c.GetMethodConfig() {
+		for _, name := range mc.GetName() {
+			n := name.GetService()
+
+			// individual method config
+			if name.GetMethod() != "" {
+				n = n + "." + name.GetMethod()
+			}
+
+			policies[n] = mc.GetRetryPolicy()
+
+			if maxReq := mc.GetMaxRequestMessageBytes(); maxReq != nil {
+				reqLimits[n] = int(maxReq.GetValue())
+			}
+
+			if maxRes := mc.GetMaxResponseMessageBytes(); maxRes != nil {
+				resLimits[n] = int(maxRes.GetValue())
+			}
+
+			if timeout := mc.GetTimeout(); timeout != nil {
+				timeouts[n] = timeout
+			}
+		}
+	}
+
+	return Config{
+		policies:  policies,
+		timeouts:  timeouts,
+		reqLimits: reqLimits,
+		resLimits: resLimits,
+	}
+}
+
+// GetRetryPolicy returns the retryPolicy and if a mapping was present for the
+// given fully-qualified Service name and simple Method name. A config mapping
+// for a specific Method takes precendence over a Service-level mapping.
+func (c Config) GetRetryPolicy(s, m string) (*MethodConfig_RetryPolicy, bool) {
```

Comment:
```suggestion
func (c Config) RetryPolicy(s, m string) (*MethodConfig_RetryPolicy, bool) {
```

https://golang.org/doc/effective_go.html#Getters (there is some extra logic, but I think the idea applies.)

Reply:
Done! Thanks for the tip.